redash/authentication/remote_user_auth.py

```diff
@@ -26,8 +29,28 @@ def login():
         email = None
 
     if not email:
-        logger.error("Cannot use remote user for login when it's not provided in the request (looked in headers['" + settings.REMOTE_USER_HEADER + "'])")
-        return redirect(url_for('redash.index', next=next_path))
+        logger.error(
+            "Cannot use remote user for login when it's not "
+            "provided in the request (looked in headers['%s'])",
+            settings.REMOTE_USER_HEADER
+        )
+        return redirect(index_url_with_next)
+
+    # Check if there is a header of user groups and if yes
+    # check it against a list of allowed user groups from the settings
+    groups = request.headers.get(settings.REMOTE_GROUPS_HEADER)
+    if groups is not None and settings.REMOTE_GROUPS_ENABLED:
+        # Check if the allowed groups match any of the ones found in the header
+        matching_groups = set()
+        for allowed_group in settings.REMOTE_GROUPS_ALLOWED:
+            matching_groups.update(set(fnmatch.filter(groups, allowed_group)))
```

Comment:
is wildcard support something we need currently?

Reply:
I don't know how the actual group notation for us is (@robotblake?) but nested groups is certainly a thing in LDAP which would make setting this stuff up simpler.

Reply:
It's mildly arbitrary, it doesn't hurt to have this though and I'm guessing while we probably won't use it right now I could see it being useful going forward.

Reply:
I changed my mind, this is a risk to land this upstream, backed it out again.